enterprise/static/enterprise/grant_data_sharing_permissions.js

```diff
@@ -1,16 +1,85 @@
+  }
+}
+
+function hideBackgroundFromTabIndex() {
+    $(".background-input").attr("tabindex", "-1");
+    $(".consent-container").attr("aria-hidden", "true");
+}
+
+function showBackgroundInTabIndex() {
+    $(".background-input").attr("tabindex", "0");
+    $(".consent-container").attr("aria-hidden", "false");
+}
+
 (function() {
-    'use strict';
+    "use strict";
 
     $(document).ready(function() {
-        $('#data-sharing').submit(function(event) {
-            var $dataSharingConsentCheckbox = $('#register-data_sharing_consent');
-            var warningMessage = $dataSharingConsentCheckbox.data('warningMessage');
-
-            if (!$dataSharingConsentCheckbox.is(':checked') && warningMessage) {
-                // eslint-disable-next-line no-alert
-                if (!window.confirm(warningMessage)) {
-                    event.preventDefault();
-                }
+        $("#failure-link").click(function (event) {
+            $("#consent-confirmation-modal").show();
+            $("#modal-close-button").focus();
+            hideBackgroundFromTabIndex();
+        });
+        $("#consent-policy-dropdown-bar").click(function (event) {
+            togglePolicy();
+        });
+        $(".policy-dropdown-link").click(function (event) {
+            togglePolicy(true);
+        });
+        $("#modal-no-consent-button").click(function (event) {
+            $("#data-consent-checkbox").attr("checked", false);
+            $("#data-sharing").submit();
+        });
+        $("#review-policy-link").click(function (event) {
+            $("#consent-confirmation-modal").hide();
+            showBackgroundInTabIndex();
+            togglePolicy(true);
+        });
+        $("#consent-confirmation-modal").click(function (event) {
+            $("#consent-confirmation-modal").hide();
+            showBackgroundInTabIndex();
+        });
+        $("#data-consent-checkbox").change(function (event) {
+            $("#consent-button").attr("disabled", !(this.checked));
+        });
+        $(document).keypress(function (event) {
+            if ((event.keyCode === 27) && ($("#consent-confirmation-modal").is(":visible"))) {
+                $("#consent-confirmation-modal").hide();
+                showBackgroundInTabIndex();
+                event.preventDefault();
+            }
+        });
+        $("#review-policy-link").keydown(function (event) {
+            if (event.keyCode == 9 && !event.shiftKey){
+                event.preventDefault();
+                $("#modal-close-button").focus();
+            }
+        });
+        $("#modal-close-button").keydown(function (event) {
+            if (event.keyCode == 9 && event.shiftKey){
+                event.preventDefault();
+                $("#review-policy-link").focus();
+            }
+        });
+        analytics.track("data_consent_form_shown");
+        analytics.trackForm($("#data-sharing"), function () {
```

Comment:
@adleyh, here's a sample of the raw payload from an `analytics.track("data_consent_denied")` call:

```json
{
  "_metadata": {
    "bundled": [
      "Segment.io"
    ],
    "unbundled": []
  },
  "anonymousId": "e4716b58-1500-4b0d-9aa0-f5754e67fad8",
  "context": {
    "campaign": {},
    "library": {
      "name": "analytics.js",
      "version": "3.0.0"
    },
    "page": {
      "path": "/enterprise/grant_data_sharing_permissions",
      "referrer": "http://localhost:8000/enterprise/grant_data_sharing_permissions?course_id=course-v1%3AedX%2BDemoX%2BDemo_Course&failure_url=http%3A%2F%2Flocalhost%3A8000%2Fdashboard%3Fconsent_failed%3Dcourse-v1%253AedX%252BDemoX%252BDemo_Course&next=http%3A%2F%2Flocalhost%3A8000%2Fcourses%2Fcourse-v1%3AedX%2BDemoX%2BDemo_Course%2Finfo",
      "search": "?course_id=course-v1%3AedX%2BDemoX%2BDemo_Course&failure_url=http%3A%2F%2Flocalhost%3A8000%2Fdashboard%3Fconsent_failed%3Dcourse-v1%253AedX%252BDemoX%252BDemo_Course&next=http%3A%2F%2Flocalhost%3A8000%2Fcourses%2Fcourse-v1%3AedX%2BDemoX%2BDemo_Course%2Finfo",
      "title": "Data sharing consent required | Your Platform Name Here",
      "url": "http://localhost:8000/enterprise/grant_data_sharing_permissions?course_id=course-v1%3AedX%2BDemoX%2BDemo_Course&failure_url=http%3A%2F%2Flocalhost%3A8000%2Fdashboard%3Fconsent_failed%3Dcourse-v1%253AedX%252BDemoX%252BDemo_Course&next=http%3A%2F%2Flocalhost%3A8000%2Fcourses%2Fcourse-v1%3AedX%2BDemoX%2BDemo_Course%2Finfo"
    },
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_3) AppleWebKit/602.4.8 (KHTML, like Gecko) Version/10.0.3 Safari/602.4.8",
    "ip": "18.189.98.227"
  },
  "event": "data_consent_denied",
  "integrations": {},
  "messageId": "ajs-277cbc68a399c54046e58fd0da58b7b0",
  "receivedAt": "2017-03-22T22:16:52.784Z",
  "sentAt": "2017-03-22T22:16:52.745Z",
  "timestamp": "2017-03-22T22:16:52.784Z",
  "type": "track",
  "userId": "4",
  "originalTimestamp": "2017-03-22T22:16:52.745Z"
}
```